### PR TITLE
add karmada performance testing process

### PR DESCRIPTION
**Pull request by Phil-sun**
Signed-off-by: Phil-sun <sunxh0000@163.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
add karmada performance testing process
**Special notes for your reviewer**:

### CODE REVIEW

File: docs/developers/karmada-performance-testing-process.md [hunk 1]
```diff
@@ -0,0 +1,588 @@
+## Karmada Performance Testing Process
```
Comment (Poor12) [inline docs/developers/karmada-performance-testing-process.md:1]: Use **front matter** to present docs:

```
---
title:
---
```

Comment (Poor12) [reply docs/developers/karmada-performance-testing-process.md:1]: As a title, I prefer ```Performance Test Setup for Karmada```

Comment (Poor12) [reply docs/developers/karmada-performance-testing-process.md:1]: For the first section, I recommend #Background.

File: docs/developers/karmada-performance-testing-process.md [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+
+**Kubemark** is directly implemented with the code of kubelet, replacing the runtime part, except that it does not actually start the container, other behaviors are exactly the same as kubelet, mainly used for Kubernetes own e2e test, simulating a large number of nodes and pods will **occupy the same memory as the real scene**.
+
+**Fake-kubelet** is a tool used to simulate any number of nodes and maintain pods on those nodes. It only does the minimum work of maintaining nodes and pods, so that it is very suitable for simulating a large number of nodes and pods for pressure testing on the control plane.
+
+#### Usage
+
+Deploy the fake-kubelet:
+```shell
+kubectl apply -f fakekubelet.yml
```
Comment (Poor12) [inline docs/developers/karmada-performance-testing-process.md:44]: Is anything different from url yamls?

**Review by Poor12 — COMMENTED** [review]
Review still going on.

File: docs/developers/performance-test-setup-for-karmada.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+namespace:
+  number: 10
+```
+
+Next, we need to specify TuningSets. TuningSet describes how actions are executed. The qps means 1/qps s per action interval. In order to distribute resources slowly to relieve the pressure on the apiserver, the qps of Uniformtinyqps is set to 0.01, which means that after distributing a deployment, we wait 10s before continuing to distribute the next deployment.
+
+```yaml
+tuningSets:
+- name: Uniformtinyqps
+    qpsLoad:
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:153]: Not aligned.

File: docs/developers/performance-test-setup-for-karmada.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+```
+
+ClusterLoader2 will create namespaces automatically, but we need to specify how many namespaces we want:
+
+```yaml
+namespace:
+  number: 10
+```
+
+Next, we need to specify TuningSets. TuningSet describes how actions are executed. The qps means 1/qps s per action interval. In order to distribute resources slowly to relieve the pressure on the apiserver, the qps of Uniformtinyqps is set to 0.01, which means that after distributing a deployment, we wait 10s before continuing to distribute the next deployment.
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:148]: 0.01 means 10s?

File: docs/developers/performance-test-setup-for-karmada.md [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+      objectTemplatePath: "deployment.yaml"
+      templateFillMap:
+        Replicas: 1000
+  - namespaceRange:
+      min: 1
+      max: 10
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    - basename: test-pp
+      objectTemplatePath: "pp.yaml"
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:182]: policy.yaml

File: docs/developers/performance-test-setup-for-karmada.md [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+    - basename: test-deployment
+      objectTemplatePath: "deployment.yaml"
+      templateFillMap:
+        Replicas: 1000
+  - namespaceRange:
+      min: 1
+      max: 10
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    - basename: test-pp
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:181]: test-policy

File: docs/developers/performance-test-setup-for-karmada.md [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+      objectTemplatePath: "pp.yaml"
+      templateFillMap:
+        Replicas: 1
+        
+```
+
+Whole `config.yaml` will look like this:
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:188]: The whole

File: docs/developers/performance-test-setup-for-karmada.md [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+
+```yaml
+name: test
+
+namespace:
+  number: 10
+
+tuningSets:
+- name: Uniformtinyqps
+    qpsLoad:
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:198]: ditto

File: docs/developers/performance-test-setup-for-karmada.md [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+    qps: 0.01
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+```
+
+Finally, we will create a phase that creates deployment and propagation policy . We need to specify in which namespaces we want the deployment and propagation policy to be created, how many of these deployments per namespace. Also, we will need to specify template for our deployment and propagation policy , which we will do later. For now, let's assume that this template allows us to specify number of replicas in deployment.
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:160]: numbers of replicas

File: docs/developers/performance-test-setup-for-karmada.md [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+
+Enter the following Prometheus expression into the ` Query` field.
+
+````sql
+histogram_quantile(0.99, sum(rate(apiserver_request_duration_seconds_bucket{verb!="WATCH|GET|LIST", resource~="works|resourcebindings"}[5m])) by (resource, verb, le))
+````
+
+The gragh will show as follow:
+
+![grafana-dashboard](../resources/contributor/grafana_metrics.png)
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:328]: move picture to /resources/adminstrator/prometheus.

File: docs/developers/performance-test-setup-for-karmada.md (hunk 9 above)
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:160]: specify in which namespaces we want the deployment and propagation policy to be created and how many of these deployments per namespace. Also, we will need to specify template for our deployment and propagation policy,

File: docs/developers/performance-test-setup-for-karmada.md [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+      min: 1
+      max: 10
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    objectBundle:
+    - basename: test-pp
+      objectTemplatePath: "pp.yaml"
+```
+
+Now, we need to specify deployment and propagation template. ClusterLoader2 by default adds parameter `Name` that you can use in your template. In our config, we also passed `Replicas` parameter. We need to remember to set correct labels so PodStartupLatency and WaitForControlledPodsRunning will watch correct pods. So our template for deployment will look like this (`deployment.yaml` file):
```
Comment (Poor12) [inline docs/developers/performance-test-setup-for-karmada.md:227]: set correct labels so PodStartupLatency and WaitForControlledPodsRunning will watch correct pods? It seems we don't test about PodStartupLatency.

**Comment by RainbowMango** [comment]
Why put resources under `adminstrator`?

**Comment by RainbowMango** [comment]
I think we should organize resources by articles.

**Review by RainbowMango — APPROVED** [review]
/approve
Thanks.
Leave LGTM to @Poor12 to confirm.

**Pull request merged**